pkg/lsp/jsonrpc2.go

```diff
@@ -0,0 +1,52 @@
+package lsp
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// ID represents a JSON-RPC 2.0 request ID, which may be either a
+// string or number (or null, which is unsupported).
+type ID struct {
```

Comment:
I add in this type rather than using `jsonrpc2.ID` to avoid this pkg having a dependency on `jsonrpc2`.